src/app/blog/[slug]/page.tsx

```diff
@@ -0,0 +1,68 @@
+import matter from "gray-matter";
+import ReactMarkdown from "react-markdown";
+import { notFound } from "next/navigation";
+import {
+  GetMarkdownFileProps,
+  MarkdownFileProps,
+  blogPostsPath,
+  generateBlogPostUrls,
+} from "./helpers";
+
+export async function generateStaticParams() {
+  const slugs = generateBlogPostUrls(blogPostsPath);
+  return slugs.map((slug: string) => {
+    return { slug };
+  });
+}
+
+export default async function BlogPost({
+  params,
+}: {
+  params: { slug: string };
+}) {
+  const { slug } = params;
+  const markdownFile = await getMarkdownFile(slug);
+
+  return (
+    <main>
+      {markdownFile?.frontmatter ? (
+        <>
+          <h1>{markdownFile.frontmatter.title}</h1>
+          <p>{markdownFile.frontmatter.date}</p>
+          <p>{markdownFile.frontmatter.tags}</p>
+          <p>{markdownFile.frontmatter.author}</p>
+          <p>Slug: {slug}</p>
```

Comment:
These could be destructured props but IIRC I'm using object notation to make TypeScript happy.